internal/request/http_test.go

```diff
@@ -101,6 +103,37 @@ func Test_http_GetUserAndGroups(t *testing.T) {
 			wantGroups:   []string{"ImpersonatedGroup"},
 			wantErr:      false,
 		},
+		{
+			name: "Certificate-ServiceAccount",
+			fields: fields{
+				Request: &http.Request{
+					Header: map[string][]string{
+						authenticationv1.ImpersonateUserHeader: {serviceaccount.ServiceAccountUsernamePrefix + "ns:account"},
+					},
+					TLS: &tls.ConnectionState{
+						PeerCertificates: []*x509.Certificate{
+							{
+								Subject: pkix.Name{
+									CommonName: serviceaccount.ServiceAccountUsernamePrefix + "ns:account",
+								},
+							},
+						},
+					},
+				},
+				authTypes: []request.AuthType{
+					request.BearerToken,
+					request.TLSCertificate,
+				},
+				client: testClient(func(ctx context.Context, obj client.Object) error {
+					ac := obj.(*authorizationv1.SubjectAccessReview)
+					ac.Status.Allowed = true
+					return nil
+				}),
+			},
+			wantUsername: serviceaccount.ServiceAccountUsernamePrefix + "ns:account",
+			wantGroups:   append([]string{fmt.Sprintf("%s%s", serviceaccount.ServiceAccountGroupPrefix, "ns")}, request.GetDefaultServiceAccountGroups()...),
```

Comment:
Same as above, I'd explicitely declare that we expect both all service accounts and authenticated groups, instead of relying on an internal global variable.